src/main/java/io/tiledb/java/api/Datatype.java

```diff
@@ -23,10 +23,55 @@
   TILEDB_STRING_UCS4,
   TILEDB_ANY;
 
+  /** @return Returns the TileDB Datatype size in Bytes * */
   public int getNativeSize() throws TileDBError {
     return tiledb.tiledb_datatype_size(this.toSwigEnum()).intValue();
   }
 
+  /** @return True if the TileDB Datatype is a scalar, false otherwise (ex. String) * */
+  public boolean isStringType() {
+    switch (this) {
+      case TILEDB_STRING_ASCII:
+      case TILEDB_STRING_UTF8:
+      case TILEDB_STRING_UTF16:
+      case TILEDB_STRING_UTF32:
+      case TILEDB_STRING_UCS2:
+      case TILEDB_STRING_UCS4:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  /** @return True if the TileDB Datatype is an integer, false otherwise * */
+  public boolean isIntegerType() {
+    switch (this) {
+      case TILEDB_CHAR:
```

Comment:
Shouldn't char be considered a string? We typically show the char as the string character and not as the ascii integer value?

Reply:
We encode char as bytes and not strings as the representation is not consistent across platforms (Char in Java is 16 bits with different semantics).